UnitTests/GitUITests/app.config

```diff
@@ -2,6 +2,10 @@
 <configuration>
   <runtime>
     <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="Microsoft.VisualStudio.Validation" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-15.3.0.0" newVersion="15.3.0.0" />
+      </dependentAssembly>
```

Comment:
Without the redirect a number of tests would fail: https://ci.appveyor.com/project/gitextensions/gitextensions/builds/21244661

```
System.IO.FileLoadException : Could not load file or assembly 'Microsoft.VisualStudio.Validation, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at GitUIPluginInterfaces.ManagedExtensibility.SetApplicationDataFolder(String applicationDataFolder)
   at GitUI.PluginRegistry.Initialize() in C:\Development\gitextensions\GitUI\Plugin\PluginRegistry.cs:line 25
   at GitUI.CommandsDialogs.FormBrowse.<<-ctor>b__56_0>d.MoveNext() in C:\Development\gitextensions\GitUI\CommandsDialogs\FormBrowse.cs:line 173
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at GitUI.CommandsDialogs.FormBrowse.<<-ctor>b__56_0>d.MoveNext() in C:\Development\gitextensions\GitUI\CommandsDialogs\FormBrowse.cs:line 179
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at GitUI.ThreadHelper.<>c__DisplayClass13_0.<<FileAndForget>b__0>d.MoveNext() in C:\Development\gitextensions\GitExtUtils\GitUI\ThreadHelper.cs:line 101
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at CommonTestUtils.ConfigureJoinableTaskFactoryAttribute.AfterTest(ITest test) in C:\Development\gitextensions\UnitTests\CommonTestUtils\ConfigureJoinableTaskFactoryAttribute.cs:line 68
   at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.Execute(TestExecutionContext context) in C:\src\nunit\nunit\src\NUnitFramework\framework\Internal\Commands\BeforeAndAfterTestCommand.cs:line 73
   at NUnit.Framework.Internal.Execution.SimpleWorkItem.PerformWork() in C:\src\nunit\nunit\src\NUnitFramework\framework\Internal\Execution\SimpleWorkItem.cs:line 59
```

It looks like the issue was somehow introduced with #5788.
I tried adding `AutoGenerateBindingRedirects` to GitUITests.csproj and other random project files, however it didn't work.

@sharwell is there a better way? Am I missing something?

Reply:
➡️ Improvement submitted as #6040